terraform-aws-static-site/main.tf

```diff
@@ -0,0 +1,163 @@
+resource "aws_s3_bucket" "non-www-bucket" {
+  bucket = var.bucket_name
+  acl    = "public-read"
+
+  website {
+    index_document = "index.html"
+    error_document = "index.html"
+  }
+}
+
+data "aws_iam_policy_document" "non-www-bucket" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [aws_cloudfront_origin_access_identity.origin_access_identity.iam_arn]
+    }
+
+    actions   = ["s3:GetObject"]
+    resources = ["arn:aws:s3:::${var.bucket_name}/*"]
+  }
+}
+
+resource "aws_cloudfront_origin_access_identity" "origin_access_identity" {}
+
+resource "aws_s3_bucket_policy" "non-www-bucket" {
+  bucket = aws_s3_bucket.non-www-bucket.id
+  policy = data.aws_iam_policy_document.non-www-bucket.json
+}
+
+resource "aws_s3_bucket" "www-bucket" {
+  bucket = "www-${var.bucket_name}"
+
+  website {
+    redirect_all_requests_to = "https://${var.redirect_address}"
+  }
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  origin {
+    domain_name = aws_s3_bucket.non-www-bucket.bucket_regional_domain_name
+    origin_id   = var.origin_id
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.origin_access_identity.cloudfront_access_identity_path
+    }
+
+    custom_origin_config {
+      origin_protocol_policy = "http-only"
+      http_port              = "80"
+      https_port             = "443"
+      origin_ssl_protocols   = ["TLSv1"]
+    }
+  }
+
+  price_class = "PriceClass_100"
+
+  enabled = true
+
+  aliases = var.aliases
+
+  default_cache_behavior {
+    allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = var.origin_id
+
+    forwarded_values {
+      query_string = true
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    compress               = true
+    min_ttl                = 0
+    default_ttl            = 31557600
+    max_ttl                = 31557600
+  }
+
+  viewer_certificate {
+    cloudfront_default_certificate = true
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  custom_error_response {
+    error_caching_min_ttl = "0"
+    error_code            = "403"
+    response_code         = "200"
+    response_page_path    = "/index.html"
+  }
+}
+
+resource "aws_cloudfront_distribution" "www_s3_distribution" {
+  origin {
+    domain_name = aws_s3_bucket.www-bucket.bucket_regional_domain_name
+    origin_id   = var.www_origin_id
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.origin_access_identity.cloudfront_access_identity_path
+    }
```

Comment:
I think we only want this origin config for the non-www cdn

Reply:
Done!